v-next/hardhat/src/internal/core/hre.ts

```diff
@@ -120,6 +121,8 @@ export class HardhatRuntimeEnvironmentImplementation
       globalOptionDefinitions,
     );
 
+    setGlobalOptionsAsEnvVariables(globalOptions);
```

Comment:
I like explicitly separating out the state side-effect here, rather than meshing it with the global option resolution.

Reply:
Okay, I have thought about this more after looking at the test.
What happens if I have a script that creates two hre instances with different `userProvidedGlobalOptions` options definitions passed in. Both instances will overwrite the ENV vars.
I think out intent is: at the top level of invocation of a task(?), convert global options to env vars, on HRE initialization it should pick up those env vars whether in a process/subprocess.